pkg/providers/pricing/pricing.go

```diff
@@ -387,11 +389,22 @@ func (p *Provider) UpdateSpotPricing(ctx context.Context) error {
 func (p *Provider) LivenessProbe(req *http.Request) error {
 	// ensure we don't deadlock and nolint for the empty critical section
 	p.mu.Lock()
-	//nolint: staticcheck
+	if err := p.checkAWSAuth(req.Context()); err != nil {
+		return err
+	}
 	p.mu.Unlock()
 	return nil
 }
 
+func (p *Provider) checkAWSAuth(ctx context.Context) error {
+	_, err := p.ec2.DescribeSpotPriceHistoryWithContext(ctx, &ec2.DescribeSpotPriceHistoryInput{DryRun: aws.Bool(true)})
+	var aerr awserr.Error
+	if errors.As(err, &aerr) && aerr.Code() == errcode.DryRunOperation {
```

Comment:
Maybe consider creating a `IsDryRunOperationError` and `IgnoreDryRunOperationError` similar to the funcs provided in the client-go errors package.